custom/src/main/java/co/elastic/otel/ElasticLogRecordExporter.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.otel;
+
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.logs.data.LogRecordData;
+import io.opentelemetry.sdk.logs.export.LogRecordExporter;
+import java.util.Collection;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class ElasticLogRecordExporter implements LogRecordExporter {
+  private static final AtomicReference<ElasticLogRecordExporter> INSTANCE = new AtomicReference<>();
+
+  private volatile boolean sendingLogs = true;
+  private final LogRecordExporter delegate;
+
+  public static ElasticLogRecordExporter getInstance() {
+    return INSTANCE.get();
+  }
+
+  static ElasticLogRecordExporter createCustomInstance(LogRecordExporter exporter) {
+    INSTANCE.set(new ElasticLogRecordExporter(exporter));
```

Comment:
For reviewer. This would ideally be a single instance that doesn't get recreated, but the test framework fails if you do that, so it needs to be recreated each time this method is called